transaction/weakset.py

```diff
@@ -76,8 +76,8 @@ def map(self, f):
     # we avoid that, although the decision to use weakrefs is now very
     # visible to our clients.
     def as_weakref_list(self):
-        # We're cheating by breaking into the internals of Python's
-        # WeakValueDictionary here (accessing its .data attribute).
-        # Python 3: be sure to freeze the list, to avoid RuntimeError:
+        # Python 3: be sure to freeze the iterator, to avoid RuntimeError:
         # dictionary changed size during iteration.
-        return list(self.data.data.values())
+        # On Python2 we already get a list, no need to copy
+        refs = self.data.valuerefs()
+        return list(refs) if PY3 else refs
```

Comment:
Since this function is so small, and called so frequently, how do people feel about just defining it differently and thus avoiding the conditional check/global access altogether? Something like:

```python
if PY3:
   def as_weakref_list(self):
      return list(self.data.valuerefs())
else:
   def as_weakref_list(self):
      return self.data.valuerefs()
```

Worth it or not?